backend/app/infrastructure/api/dependencies.py

```
@@ -0,0 +1,90 @@
+from app.domain.interfaces.user_repository import IUserRepository
+from app.infrastructure.persistence.dynamodb import (
+    DynamoDBFeedbackRepository,
+    DynamoDBGroceryListRepository,
+    DynamoDBMealPlanRepository,
+    DynamoDBRecipeRepository,
+    DynamoDBUserRepository,
+)
+
+
+@lru_cache(maxsize=1)
+def get_recipe_repository(settings: Settings = Depends(get_settings)) -> IRecipeRepository:
```

Ruff (0.14.13)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)